name: All_Together

on:
  # push:
  #   branches: [ "main" ]
  pull_request:
    branches: [ "main" ]
  workflow_dispatch:
env:
  file_to_build: all_together.py
  pi_image: raspios_lite_arm64:latest 
  path_to_file: IIoT_Device_Guide_UGA_2023/all_together
  built_file_name: all_together


jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - name: Checkout repo
      uses: actions/checkout@main
      with: 
        lfs: true
    - name: checkout lfs object
      run: git lfs checkout
    - name: Build test
      uses: pguyot/arm-runner-action@main
      id: Build-Python-exe
      with:
        image_additional_mb: 3584
        copy_repository_path: /home/pi
        copy_artifact_path: ${{env.path_to_file}}/dist/${{env.built_file_name}}
        base_image: ${{env.pi_image}}
        commands: |
            sudo apt-get update -y;
            pwd;
            apt-get install -y python3 python3-venv python3-dev python3-pip; 
            cd ${{env.path_to_file}};
            pip3 install requests;
            pip3 install --upgrade pip;
            pip3 install time;
            pip3 install random;
            pip3 install zlib;
            pip3 install pyinstaller;
            pip3 install hashlib;
            pip3 install mmh3;
            pip3 install math;
            pyinstaller --clean --onefile ${{env.file_to_build}};
            ls -a;
            cd ..
            sudo cp /home/pi/IIoT_Device_Guide_UGA_2023/startup/all_together.service /lib/systemd/system/all_together.service
            sudo chmod 644 /lib/systemd/system/all_together.service
            sudo systemctl enable all_together.service
    - uses: actions/upload-artifact@main
      with:
        name: exe_file
        path: ${{env.built_file_name}}
        if-no-files-found: error
        retention-days: 3

